dags/js/function-caller.js

```diff
@@ -87,5 +87,10 @@ module.exports = functionCaller;
 // if this script has been called directly then run the code in the if block
 // equivalent to python's __name__ == '__main__' comparison
 if (!module.parent) {
-  functionCaller();
+  try {
+    functionCaller();
```

Comment:
`functionCaller` is `async` but we can't `await` here since we are at the top level of the module. The `catch` may not see errors then, will have to cross-check the approach

Reply:
Can confirm that errors are definitely caught here. You can try this locally by throwing an error somewhere in the code.